dbt_platform_helper/providers/schema_migrations.py

```diff
@@ -0,0 +1,106 @@
+from collections import Counter
+from copy import deepcopy
+from typing import Protocol
+
+from dbt_platform_helper.platform_exception import PlatformException
+
+
+class InvalidMigrationConfigurationException(PlatformException):
+    pass
+
+
+class PlatformConfigSchemaMigrationException(PlatformException):
+    pass
+
+
+class SchemaMigrationProtocol(Protocol):
+    def from_version(self) -> int: ...
+
+    def migrate(self, platform_config: dict) -> dict: ...
+
+    def messages(self) -> list[str]: ...
+
+
+class SchemaV0ToV1Migration:
+    def __init__(self):
+        self._messages = []
+
+    def from_version(self) -> int:
+        return 0
+
+    def migrate(self, platform_config: dict) -> dict:
+        migrated_config = deepcopy(platform_config)
+
+        self._remove_terraform_platform_modules_default_version(migrated_config)
+        self._remove_versions_from_env_config(migrated_config)
+        self._remove_to_account_and_from_account_from_database_copy(migrated_config)
+
+        return migrated_config
+
+    def messages(self):
+        return self._messages
+
+    def _remove_versions_from_env_config(self, migrated_config: dict) -> None:
+        for env_name, env in migrated_config.get("environments", {}).items():
+            if env and "versions" in env:
+                del env["versions"]
+                self._messages.append(
+                    f"environments.{env_name}.versions is deprecated and no longer used. Please delete from your platform-config.yml"
```

Comment:
Not sure about this message, didn't we just delete the property above?

Same question applies below I think.

Reply:
Yes. This is a hangover from when I was "automagically" updating the platform-config on the fly... Good spot.